### DISCUSSION TITLE
prisma schema validation

### DISCUSSION THREAD

**Discussion by Yesh-hacker** [prisma/prisma#25701] [Q&A]
### Question

when I ran this 
```
npm exec prisma migrate dev
```
I faced this error 
```
{
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database

Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Environment variable not found: DIRECT_URL.
  -->  prisma\schema.prisma:4
   |
 3 |   url       = env("DATABASE_URL")
 4 |   directUrl = env("DIRECT_URL")
   |

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 5.22.0
}
```
### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Yesh-hacker 

From the error message, the `DIRECT_URL` environment variable appears missing or undefined in your `.env` file.  Can you confirm that you have defined `DIRECT_URL` in your `.env` file?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
